src/hotspot/share/prims/jvmtiRedefineClasses.cpp

```diff
@@ -1700,14 +1671,6 @@ bool VM_RedefineClasses::merge_constant_pools(const constantPoolHandle& old_cp,
       if (match) {
         // found a match at the same index so nothing more to do
         continue;
-      } else if (is_unresolved_class_mismatch(scratch_cp, scratch_i,
-                                              *merge_cp_p, scratch_i)) {
-        // The mismatch in compare_entry_to() above is because of a
-        // resolved versus unresolved class entry at the same index
-        // with the same string value. Since Pass 0 reverted any
-        // class entries to unresolved class entries in *merge_cp_p,
-        // we go with the unresolved class entry.
-        continue;
```

Comment:
I'm sorry for the piecemeal review.   There's another comment that mentions this comment that should be removed.  It starts with this:

-      // The find_matching_entry() call above could fail to find a match